toqito/matrix_props/is_doubly_stochastic.py

```diff
@@ -0,0 +1,75 @@
+"""Check is a matrix is doubly stochastic."""
+
+import numpy as np
+
+from toqito.matrix_props import is_left_stochastic, is_right_stochastic
+
+
+def is_doubly_stochastic(mat: np.ndarray) -> bool:
+    r"""Verify matrix is doubly stochastic.
+
+    A matrix is doubly stochastic if it is also a right and left stochastic matrix.
+    :cite:`WikiStichasticMatrix, WikiDoublyStichasticMatrix`.
+
+    See Also
+    ========
+    is_right_stochastic
+    is_left_stochastic
+
+    Examples
+    ========
+    The elements an identity matrix and a Pauli X matrix are nonnegative where both the rows and columns sum up to 1.
+    The same cannot be said about a Pauli Z matrix.
+
+    .. math::
+    Id = \begin{pmatrix}
+            1 & 0 & 0 & 0 & 0\\
+            0 & 1 & 0 & 0 & 0\\
+            0 & 0 & 1 & 0 & 0\\
+            0 & 0 & 0 & 1 & 0\\
+            0 & 0 & 0 & 0 & 1\\
+        \end{pmatrix}
+
+    >>> import numpy as np
+    >>> from toqito.matrix_props import is_doubly_stochastic
+    >>> id_mat = np.eye(5)
```

Comment:
We don't need to define a separate variable for the identity, we can just use `np.eye(5)` directly.